guides/micronaut-cloud-oidc-oracle/micronaut-cloud-oidc-oracle.adoc

```diff
@@ -0,0 +1,178 @@
+Click the image:cloud-oidc-oracle/burgermenu.png[] ("burger" menu) in the top left, then click "Settings", and then "Default Settings":
+
+image::cloud-oidc-oracle/federation16.png[]
+
+and enable unauthenticated access to the signing certificate:
+
+image::cloud-oidc-oracle/federation17.png[]
+
+common:create-app-features.adoc[]
+
+=== Configuration
+
+Replace the generated `application.yml` with this:
+
+resource:application.yml[]
+
+<1> Set `micronaut.security.authentication` as `idtoken`. The `idtoken` provided by your OAuth 2.0 application when the  Authorization code flow ends (a signed JWT) will be saved in a cookie.
+callout:oauth2-client-name[number=2,arg0=oci]
+<3> `client-id`, `client-secret`, and the `issuer` URL will be set as environment variables
+<4> Accept GET requests to the `/logout` endpoint
+
+=== Home controller
+
+Create a `HomeController` class to test the security configuration:
+
+source:HomeController[]
+
+callout:controller[number=1,arg0=/]
+callout:secured-anonymous[number=2]
+callout:view[number=3]
+callout:get[number=4,arg0=index,arg1=/]
+callout:secured-is-authenticated[number=5]
+callout:get[number=6,arg0=secured,arg1=/secure]
+<7> This method simply returns a model map that will be rendered   as JSON since there's no `@View` annotation
+
+=== Thymeleaf view
+
+Create a Thymeleaf view to render the UI for the controller:
+
+resource:views/home.html[]
+
+Note that we return an empty model in the controller `index()` method, however we are accessing `security` in the Thymeleaf template.
+The Micronaut Security https://micronaut-projects.github.io/micronaut-views/latest/api/io/micronaut/views/model/security/SecurityViewModelProcessor.html[SecurityViewModelProcessor]
+injects a `security` map into the model with the authenticated user data.
+See the https://micronaut-projects.github.io/micronaut-views/latest/guide/#security-model-enhancement[User in a view] documentation.
+
+== Running the Application
+
+Before starting the application we need to create some environment variables.
+We used placeholders in `application.yml` for the `client-id`, `client-secret`, and the `issuer` URL properties, so we need to create `OAUTH_CLIENT_ID`, `OAUTH_CLIENT_SECRET`, and `OAUTH_ISSUER` environment variables.
+Use the client ID and client secret you saved when you created the Oracle Cloud application, and the issuer URL you saved (the root of the Oracle Identity Cloud Service (OICS) console URL):
+
+[source, bash]
+----
+export OAUTH_CLIENT_ID=XXXXXXXXXX
+export OAUTH_CLIENT_SECRET=YYYYYYYYYY
+export OAUTH_ISSUER=https://idcs-7084de14ec.......identity.oraclecloud.com/
+----
+
+common:env-vars.adoc[]
+
+common:runapp-instructions.adoc[]
+
+Open http://localhost:8080/secure in a browser and you'll be redirected to the home page because the controller method is annotated with `@Secured(IS_AUTHENTICATED)` and we're not yet authenticated:
```

Comment:
```suggestion
Open http://localhost:8080/secure in a browser, and you'll be redirected to the home page because the controller method is annotated with `@Secured(IS_AUTHENTICATED)`, and you're not yet authenticated:
```